python/ray/air/integrations/wandb.py

```diff
@@ -0,0 +1,276 @@
+    wandb.data_types.Table,
+    wandb.data_types.Video,
+)
+_VALID_ITERABLE_TYPES = (
+    wandb.data_types.Audio,
+    wandb.data_types.BoundingBoxes2D,
+    wandb.data_types.Graph,
+    wandb.data_types.Histogram,
+    wandb.data_types.Html,
+    wandb.data_types.Image,
+    wandb.data_types.ImageMask,
+    wandb.data_types.Molecule,
+    wandb.data_types.Object3D,
+    wandb.data_types.Plotly,
+    wandb.data_types.Table,
+    wandb.data_types.Video,
+)
+
+
+_MockWandb = MagicMock
+
+
+def setup_wandb(config: Optional[Dict] = None, rank_zero_only: bool = True, **kwargs):
+    """Set up a Weights & Biases session.
+
+    This function can be used to initialize a Weights & Biases session in a
+    (distributed) training or tuning run.
+
+    Per default, the run ID is the trial ID, the run name is the trial name, and
+    the run group is the experiment name. These settings can be overwritten by
+    passing the respective arguments as ``kwargs``, which will be passed to
+    ``wandb.init()``.
+
+    In distributed training with Ray Train, only the zero-rank worker will initialize
+    wandb. All other workers will return a mock object, so that logging is not
+    duplicated in a distributed run. This can be disabled by passing
+    ``rank_zero_only=False``, which will then initialize wandb in every training
+    worker.
+
+    The ``config`` argument will be passed to Weights and Biases and will be logged
+    as the run configuration. If wandb-specific settings are found, they will
+    be used to initialize the session. These settings can be
+
+    - api_key_file: Path to locally available file containing a W&B API key
+    - api_key: API key to authenticate with W&B
+
+    If no API information is found in the config, wandb will try to authenticate
+    using locally stored credentials, created for instance by running ``wandb login``.
+
+    All other keys found in the ``wandb`` config parameter will be passed to
+    ``wandb.init()``. If the same keys are present in multiple locations, the
+    ``kwargs`` passed to ``setup_wandb()`` will take precedence over those passed
+    as config keys.
+
+    Args:
+        config: Configuration dict to be logged to weights and biases. Can contain
+            arguments for ``wandb.init()`` as well as authentication information.
+        rank_zero_only: If True, will return an initialized session only for the
+            rank 0 worker in distributed training. If False, will initialize a
+            session for all workers.
+        kwargs: Passed to ``wandb.init()``.
+
+    Example:
+
+        .. code-block: python
+
+            from ray.air.integrations.wandb import wandb_setup
+
+            def training_loop(config):
+                wandb = wandb_setup(config)
+                # ...
+                wandb.log({"loss": 0.123})
+
+    """
+    try:
+        # Do a try-catch here if we are not in a train session
+        if rank_zero_only and session.get_local_rank() != 0:
```

Comment:
The API for logging uses the `wandb` module, not the run object right? 

it doesn't seem like there's a way for us to handle the "log only on rank 0" for the user.